kev1n1in/AsiaYoTravelMap
# Feature/homepage

[WebUrl](https://asiayo-e20a4--kevin-ta7gp02h.web.app)

## Description

1. Background Update for Login Page:

- Changed the background of the login page to a video background, enhancing visual appeal and user experience.
- The video background is sourced from https://coverr.co/.

2. Component Refactoring:

- Refactored Dialog and Alert messages into separate components for better reusability and maintainability.
- AlertMessage can now be used across the application, not just on the login page.

3. Login Logic Extraction:

- Extracted login logic into the GoogleLogin component, making the Login component more focused on display and state management.
- GoogleLogin handles the Google login process and passes user data to the Login component through the onLoginSuccess callback.

## Test Plan

1. Video Background Check:

- Verify that the login page displays the correct video background.
- Ensure the video auto-plays, is muted, and loops correctly.

2. Dialog and Alert Components:

- Test that the AlertMessage component correctly displays error messages.
- Ensure that the AlertMessage component shows and hides appropriately as needed.


3. GoogleLogin Component:

- Verify that the GoogleLogin component functions correctly for the Google login process.
- Ensure that the onLoginSuccess callback is called correctly after a successful login, and user data is correctly set in the Login component.
- Test that error messages are displayed correctly if an issue occurs during the login process.

4. Login Logic:

- Test that after a successful login, the application correctly redirects to the /home page.
- Ensure the logout functionality works correctly and that the user is redirected to the homepage after logging out.

5. Responsive Design:

- Verify that the login page displays correctly on different devices and screen sizes.
- Check that the video background and buttons display properly across various resolutions.
